### PR TITLE
perf!: remove DeepRequired from sanitized config types

### PR DESCRIPTION
**Pull request by AlessioGr**
Follow up of https://github.com/payloadcms/payload/pull/17541

This PR removes `DeepRequired` from `SanitizedConfig`. It uses shallow `Pick`, `Omit`, and `Required` types so only properties populated during root config sanitization are required.

## Why

`DeepRequired` recursively processes large admin, field, hook, GraphQL, and plugin types, adding unnecessary TypeScript checker work.

It was also inaccurate. `admin.autoLogin`, `admin.livePreview`, `graphQL.mutations`, `graphQL.queries`, `queryPresets`, and `typescript.postProcess` could be undefined but were typed as required. Meanwhile, `blocks` is always populated, `auth.jwtOrder` is supplied by Payload despite previously being required in the incoming `Config`, and `SanitizedConfig.paths` was typed but never existed at runtime.

The runtime defaults now also preserve the guarantees made by `SanitizedConfig`: explicitly passing `undefined` cannot erase required admin, GraphQL, route, TypeScript, or auth defaults, and localization always receives `localeCodes` and `fallback` when enabled.

## Breaking change

Code using `SanitizedConfig` may now need to handle `undefined` for properties Payload does not default. `Config.auth.jwtOrder` is now optional for incoming configs, and the unused `SanitizedConfig.paths` property has been removed.

### CODE REVIEW

File: packages/payload/src/config/sanitize.ts [hunk 1]
```diff
@@ -267,7 +267,7 @@ export const sanitizeConfig = (incomingConfig: Config): SanitizedConfig => {
     config.endpoints.push(endpoint)
   }
 
-  if (config.localization && config.localization.locales?.length > 0) {
+  if (config.localization) {
```
Comment (AlessioGr) [inline packages/payload/src/config/sanitize.ts:270]: `SanitizedLocalizationConfig` guarantees `localeCodes` and `fallback`, but the old condition skipped populating them when `locales: []`.

File: packages/payload/src/admin/views/index.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ import type { Data, StaticDescription } from '../types.js'
 import type { DocumentSubViewTypes } from './document.js'
 
 export type AdminViewConfig = {
-  Component: PayloadComponent
+  Component: PayloadComponent<AdminViewServerProps>
```
Comment (AlessioGr) [inline packages/payload/src/admin/views/index.ts:22]: Removing `DeepRequired` preserves the real `PayloadComponent` generic and exposed that admin views were typed too loosely. This explicitly records the server props the component _already receives_.

File: packages/payload/src/config/defaults.ts [hunk 3]
```diff
@@ -4,35 +4,38 @@ import { defaultAccess } from '../auth/defaultAccess.js'
 import { databaseKVAdapter } from '../kv/adapters/DatabaseKVAdapter.js'
 
 export const addDefaultsToConfig = (config: Config): Config => {
+  const admin = config.admin
+
   config.admin = {
-    avatar: 'gravatar',
```
Comment (AlessioGr) [inline packages/payload/src/config/defaults.ts:8]: Spread first. Ensures they cannot be later assigned to `undefined`, which would break the SanitizedConfig type contract which marks these as required

File: packages/payload/src/uploads/cropImage.ts [hunk 4]
```diff
@@ -17,7 +17,7 @@ type CropImageArgs = {
   file: PayloadRequest['file']
   heightInPixels: number
   req?: PayloadRequest
-  sharp: SanitizedConfig['sharp']
+  sharp: NonNullable<SanitizedConfig['sharp']>
```
Comment (AlessioGr) [inline packages/payload/src/uploads/cropImage.ts:20]: `sharp` is optional globally, but `cropImage` is only called after checking it exists

File: packages/payload/src/utilities/headersWithCors.ts [hunk 5]
```diff
@@ -29,7 +29,7 @@ export const headersWithCors = ({ headers, req }: CorsArgs): Headers => {
       if (typeof cors === 'object' && 'headers' in cors) {
         headers.set(
           'Access-Control-Allow-Headers',
-          [...defaultAllowedHeaders, ...cors.headers].filter(Boolean).join(', '),
+          [...defaultAllowedHeaders, ...(cors.headers ?? [])].filter(Boolean).join(', '),
```
Comment (AlessioGr) [inline packages/payload/src/utilities/headersWithCors.ts:32]: `cors.headers` is optional. `DeepRequired` previously hid that, and spreading undefined could throw when only `origins` was configured. The empty-array fallback preserves the default headers safely.

File: packages/payload/src/config/sanitize.ts (hunk 1 above)
Comment (paulpopus) [inline packages/payload/src/config/sanitize.ts:270]: This line now runs the block whenever config.localization is set. Before this change, the block skipped configs with an empty locales array. An empty locales array now gets default localeCodes and fallback values instead of undefined. Add a test that sanitizes a config with an empty locales array and checks that result.

**Review by paulpopus — CHANGES_REQUESTED** [review]
SanitizedGlobalConfig (packages/payload/src/globals/config/types.ts) still extends DeepRequired<GlobalConfig>. Global config sanitization keeps the same accuracy problem that this PR fixes for the root config. Confirm if global config work is planned for a follow-up PR, or include it in this PR.

There are no unit tests for sanitizeConfig or addDefaultsToConfig in packages/payload/src/config. This PR depends on manual tracing to confirm which properties sanitization always sets. Add characterization tests that check the sanitized shape of admin, graphQL, typescript, and routes for a minimal input config.

File: packages/payload/src/config/sanitize.ts (hunk 1 above)
Comment (AlessioGr) [reply packages/payload/src/config/sanitize.ts:270]: Done! Added tests

**Comment by AlessioGr** [comment]
> SanitizedGlobalConfig (packages/payload/src/globals/config/types.ts) still extends DeepRequired. Global config sanitization keeps the same accuracy problem that this PR fixes for the root config. Confirm if global config work is planned for a follow-up PR, or include it in this PR.

For the sake of reviewability, remaining usage of DeepRequired will be addressed in a future PR!

File: packages/payload/src/config/sanitize.ts [hunk 6]
```diff
@@ -267,7 +267,7 @@ export const sanitizeConfig = (incomingConfig: Config): SanitizedConfig => {
...
     // clone localization config so to not break everything
     const firstLocale = config.localization.locales[0]
```
Comment (paulpopus) [inline packages/payload/src/config/sanitize.ts:272]: This code reads config.localization.locales[0] without a check. The previous condition skipped this block when locales was missing or empty. A config that omits the locales array now throws an unhandled type error instead of a clear message. Add a check for Array.isArray(config.localization.locales) before this line, and throw an InvalidConfiguration error when the check fails.

File: packages/payload/src/config/sanitize.spec.ts [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      autoGenerate: true,
+      outputFile: `${process.cwd()}/payload-types.ts`,
+    })
+    expect(sanitizedConfig.routes).toEqual({
+      admin: '/admin',
+      api: '/api',
+      graphQL: '/graphql',
+      graphQLPlayground: '/graphql-playground',
+    })
+  })
```
Comment (paulpopus) [inline packages/payload/src/config/sanitize.spec.ts:74]: The new tests check defaults for a minimal config and for an empty locales array. They do not check a config that sets a nested field to undefined, for example admin.avatar. The defaults.ts fix in this PR depends on this exact case working. Add a test that sets admin.avatar to undefined and confirms the sanitized config still returns the default value.

**Review by paulpopus — CHANGES_REQUESTED** [review]

File: packages/payload/src/config/sanitize.ts (hunk 6 above)
Comment (AlessioGr) [reply packages/payload/src/config/sanitize.ts:272]: > A config that omits the locales array now throws an unhandled type error instead of a clear message

Our types define the `locales` property as required, so I think this is fine and not worth additional runtime check. In the distant past, we used to use `joi` to validate the config during runtime, but made the decision that type-checking is enough. Same scenario here - if we want to perform a runtime validation check, we should hold the entire config to the same standard (= introduce `joi` or `zod`), which would be out of scope for this PR

File: packages/payload/src/config/sanitize.spec.ts (hunk 7 above)
Comment (AlessioGr) [reply packages/payload/src/config/sanitize.spec.ts:74]: done!

**Review by paulpopus — APPROVED** [review]
